kayttooikeus-service/src/main/java/fi/vm/sade/kayttooikeus/util/HenkilohakuBuilder.java

```diff
@@ -70,10 +80,22 @@ public HenkilohakuBuilder search(Long offset, OrderByHenkilohaku orderBy) {
 
     // Remove henkilos the user has no access (and who have organisation)
     public HenkilohakuBuilder exclusion() {
-        if(!this.permissionCheckerService.isCurrentUserAdmin()) {
+        if (!this.permissionCheckerService.isCurrentUserAdmin()) {
+            List<String> organisaatioOids = this.organisaatioHenkiloRepository.findDistinctOrganisaatiosForHenkiloOid(this.permissionCheckerService.getCurrentUserOid());
 
-        }
+            if (!organisaatioOids.contains(this.commonProperties.getRootOrganizationOid())) {
+                if (henkilohakuCriteriaDto.getOrganisaatioOids() == null) {
+                    henkilohakuCriteriaDto.setOrganisaatioOids(organisaatioOids);
+                } else {
+                    List<String> allCurrentUserOrganisaatioOids = organisaatioOids.stream()
+                            .flatMap(currentUserOrganisaatioOid -> this.organisaatioClient.getChildOids(currentUserOrganisaatioOid).stream())
+                            .collect(Collectors.toList());
 
+                    allCurrentUserOrganisaatioOids.retainAll(henkilohakuCriteriaDto.getOrganisaatioOids());
+                    henkilohakuCriteriaDto.setOrganisaatioOids(allCurrentUserOrganisaatioOids);
+                }
+            }
```

Comment:
Mietin, että pystyykö näitä kaikkia henkilöitä jotka ovat samassa tai aliorganisaatiossa edes näkemään virkailja näkymästä mutta mennään nyt speksin mukaan.